alf/algorithms/actor_critic_algorithm.py

```diff
@@ -128,6 +139,8 @@ def __init__(self,
             loss = loss_class(debug_summaries=debug_summaries)
         self._loss = loss
 
+        self._post_init()
```

Comment:
The initial commit implemented auto call: ee8f83e266e452bac3a62872f436c56cc679fed0
Unfortunately, there is a confliction with alf.config (since alf config uses signatures), which is the reason CI failed. 
Also tried another two different approaches for also call but have the same issue. 

Please let me know if you have any suggestions on implementing this auto call.

Reply:
> The initial commit implemented auto call: [ee8f83e](https://github.com/HorizonRobotics/alf/commit/ee8f83e266e452bac3a62872f436c56cc679fed0) Unfortunately, there is a confliction with alf.config (since alf config uses signatures), which is the reason CI failed. Also tried another two different approaches for also call but have the same issue.
> 
> Please let me know if you have any suggestions on implementing this auto call.

Just traverse the entire algorithm tree and call pre-loading for each algorithm at the first line of `PolicyTrainer.train()`?

Reply:
> > The initial commit implemented auto call: [ee8f83e](https://github.com/HorizonRobotics/alf/commit/ee8f83e266e452bac3a62872f436c56cc679fed0) Unfortunately, there is a confliction with alf.config (since alf config uses signatures), which is the reason CI failed. Also tried another two different approaches for also call but have the same issue.
> > Please let me know if you have any suggestions on implementing this auto call.
> 
> Just traverse the entire algorithm tree and call pre-loading for each algorithm at the first line of `PolicyTrainer.train()`? Similar for `PolicyTrainer.play()`.

It's doable, but there might be some other use cases not covered by this approach.  E.g. someone simply constructed an algorithm instance with checkpoint arg specified, as in the test case.

Reply:
> > > The initial commit implemented auto call: [ee8f83e](https://github.com/HorizonRobotics/alf/commit/ee8f83e266e452bac3a62872f436c56cc679fed0) Unfortunately, there is a confliction with alf.config (since alf config uses signatures), which is the reason CI failed. Also tried another two different approaches for also call but have the same issue.
> > > Please let me know if you have any suggestions on implementing this auto call.
> > 
> > 
> > Just traverse the entire algorithm tree and call pre-loading for each algorithm at the first line of `PolicyTrainer.train()`? Similar for `PolicyTrainer.play()`.
> 
> It's doable, but there might be some other use cases not covered by this approach. E.g. someone simply constructed an algorithm instance with checkpoint arg specified, as in the test case.

It's just like `compile()` in hobot. The point is that, when needed, we only need to call once on the root algorithm.

Reply:
Now used auto post init call, which will handle checkpoint loading automatically.